### ISSUE TITLE
"Update" on JSON field preserves the old KV pairs

### ISSUE THREAD

**Issue by Redzli** [prisma/prisma-client-js#790]
Hey, when I run an `update` statement on a table that has a JSON field, I would like to `update` the JSON field that has some existing data, instead of "replacing" it. What I mean by this is to append another key-value pair to the JSON field, while preserving the old data in the JSON.

My schema: 
```prisma
model User {
  userMetadata   Json?
```

When I created the user, I did
```javascript
const createUser = await prisma.user.create({
  data: {
    userMetadata: {
       name: "John"
    }
  }
});
```

Then when I want to update his metadata, I do:

```javascript
  await prisma.user.update({
    where: {
      id: userId //suppose there is one
    },
    data:{
      userMetadata: {
        university: "Oxford"
      }
    }
  })
  ```

## Expected results
Ideally after running the `update` I would like to see the user having such `userMetadata`:
```
userMetadata: {
  name: "John",
  university: "Oxford"
}
```

## Actual results
However, it only shows 

```
userMetadata: {
  university: "Oxford"
}
```

## Question
I am wondering if there is a way to append new key-value pairs like `university: "Oxford"` to the userMetadata while preserving the old data?

**Comment by janpio** [comment]
Right now `update` is fully replacing the value of your `userMetadata` column by design. You would have to read the data, modify it yourself and then write it back to achieve what you are looking for.

**Comment by tre-dev** [comment]
While it works by reading the data and `Object.assign(existing, new)`, this creates an overhead of having to read the data first. I wouldn't say it's an important feature to have ASAP, but most def. something to improve latency & reduce overhead.

**Comment by Redzli** [comment]
Thanks for the answer. I will use the workaround as you suggest. I understand while it is still a new feature awaiting implementation, you won't work on it any time soon. For the sake of this ticket, shall I close it, or leave it open?

**Comment by janpio** [comment]
No, unless someone points out a better issue, this is now the issue tracking this feature being wished for and later prioritized and implemented. Improving the Json behavior is definitely on our list.
